OneSignalSDK/onesignal/notifications/src/main/java/com/onesignal/notifications/internal/common/OSWorkManagerHelper.kt

```diff
@@ -0,0 +1,46 @@
+package com.onesignal.notifications.internal.common
+
+import android.annotation.SuppressLint
+import android.content.Context
+import androidx.work.Configuration
+import androidx.work.WorkManager
+import androidx.work.impl.WorkManagerImpl
+
+object OSWorkManagerHelper {
+    /**
+     * Helper method to provide a way to check if WorkManager is initialized in this process.
+     * The purpose of this helper is to work around this bug - https://issuetracker.google.com/issues/258176803
+     *
+     * This is effectively the `WorkManager.isInitialized()` public method introduced in androidx.work:work-*:2.8.0-alpha02.
+     * Please see https://android-review.googlesource.com/c/platform/frameworks/support/+/1941186.
+     *
+     * @return `true` if WorkManager has been initialized in this process.
+     */
+    @SuppressWarnings("deprecation")
+    @SuppressLint("RestrictedApi")
+    private fun isInitialized(): Boolean {
+        return WorkManagerImpl.getInstance() != null
+    }
+
+    /**
+     * If there is an instance of WorkManager available, use it. Else, in rare cases, initialize it ourselves.
+     *
+     * Calling `WorkManager.getInstance(context)` directly can cause an exception if it is null.
+     *
+     * @return an instance of WorkManager
+     */
+    @JvmStatic
+    @Synchronized
+    fun getInstance(context: Context): WorkManager {
+        if (!isInitialized()) {
+            try {
+                // Initialization can fail if another thread initializes in the small time gap
+                // https://android.googlesource.com/platform/frameworks/support/+/60ae0eec2a32396c22ad92502cde952c80d514a0/work/workmanager/src/main/java/androidx/work/impl/WorkManagerImpl.java#177
+                WorkManager.initialize(context, Configuration.Builder().build())
+            } catch (e: IllegalStateException) {
+                // Admittedly starting to get hacky
```

Comment:
I would recommend we print something here, it could be helpful to know what `e` is here if we ever hit it.

Also, I believe we would probably only get here in two cases:
1. We lost the race with another call to ` WorkManager.initialize` outside of OneSignal.
2. If some other expected error is thrown from WorkManager.

I would replace your "Admittedly starting to get hacky" comment with something like that above, assuming I got all the details right.

Reply:
Changed! The added try-catch was for this [comment](https://github.com/OneSignal/OneSignal-Android-SDK/issues/1672#issuecomment-1720744291) after the player model fix. 

I added the log. The original issue of WorkManager not initialized is already so rare, it would be even rarer to then reproduce failing on its initialization, but it's possible.